internal/controller/replicationgroupsource_controller.go

```diff
@@ -129,13 +139,24 @@ func (r *ReplicationGroupSourceReconciler) Reconcile(ctx context.Context, req ct
 
 // SetupWithManager sets up the controller with the Manager.
 func (r *ReplicationGroupSourceReconciler) SetupWithManager(mgr ctrl.Manager) error {
-	return ctrl.NewControllerManagedBy(mgr).
+	vgsCRD := &apiextensionsv1.CustomResourceDefinition{}
+	if err := r.Client.Get(context.TODO(),
```

Comment:
IIRC, you cannot use r.Client for this call. You must use the APIReader client as the r.Client is a cached client and it only builds the caching after starting the manager.

Reply:
fixed